pkg/storage/ingest/reader.go

```diff
@@ -955,12 +990,23 @@ func newConcurrentFetchers(
 	}
 	f.topicID = topics[topic].ID
 
-	go f.runFetchers(ctx, startOffset)
+	go f.start(ctx, startOffset, concurrency, recordsPerFetch)
 
 	return f, nil
 }
 
-func (r *concurrentFetchers) pollFetches(ctx context.Context) (kgo.Fetches, context.Context) {
+func (r *concurrentFetchers) Update(ctx context.Context, concurrency, records int) {
+	if r.concurrency == concurrency && r.recordsPerFetch == records {
```

Comment:
why do this check? i think we can just get rid of r.concurrency and r.recordsPerFetch; subsequent calls to Update would be buggy

Reply:
https://github.com/grafana/mimir/pull/9437/commits/918256ced402839c2cc59012c1115d90c870e780